jpos/src/main/java/org/jpos/iso/packager/JSONPackager.java

```diff
@@ -87,9 +93,26 @@ private void put (Map map, ISOComponent c, String prefix) throws ISOException {
                 put (map, cc, c.getFieldNumber() > 0 ? prefix + Integer.toString(c.getFieldNumber()) + "." : prefix);
             }
         }
-        else if (c instanceof ISOField)
-            map.put(prefix + c.getKey(), ((ISOField)c).getValue());
-        else if (c instanceof ISOBinaryField)
+        else if (c instanceof ISOField){
+            if (packager==null){
+                map.put(prefix + c.getKey(), ((ISOField)c).getValue());
+            }else {
+                Map field = new LinkedHashMap();
+                if((((ISOBasePackager) packager).getFieldPackager((int) c.getKey()).getClass().getName()).contains("NUMERIC")) {
```

Comment:
I wouldn't go by name containing NUMERIC to determine the type - need something better - isn't there some packager patterns JPOS uses already?

Reply:
Hey Vishu, I didn't merge this PR just yet because I'm not sure which JSON dependency we want to add. In jPOS-EE we are using Jackson, I believe this one uses JSON Simple (kinda dead), we have some others using GSON (kinda dead). @vsalaman suggested  https://bolerio.github.io/mjson which looks very nice. So we'll see which one to use in jPOS, and then rework this PR with @andresantoniuk

Reply:
@ar thanks for the note. I didn't know GSON is considered obsolete. I see commits in Aug 2018. I haven't used mjson but will try it out some time.